File: SkaiLab.Invoice/ClientApp/src/app/models/product-sale-summary.ts
import { Product } from "./product";

export class ProductSaleSummary{
    product:Product;
    avgPurchasePrice:number;
    purchaseQty:number;
    purchaseTotal:number;
    avgSalePrice:number;
    saleQty:number;
    saleTotal:number;
    netQty:number;
    netTotal:number;
    constructor(){
        this.product=new Product();
        this.avgPurchasePrice=0;
        this.purchaseQty=0;
        this.purchaseTotal=0;
        this.avgSalePrice=0;
        this.saleQty=0;
        this.saleTotal=0;
        this.netQty=0;
        this.netTotal=0;
    }
}

File: SkaiLab.Invoice.Models/Report/ProductSaleSummary.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SkaiLab.Invoice.Models.Report
{
    public class ProductSaleSummary
    {
        public Product Product { get; set; }
        public decimal AvgPurchasePrice { get; set; }
        public int PurchaseQty { get; set; }
        public decimal PurchaseTotal { get; set; }
        public decimal AvgSalePrice { get; set; }
        public int SaleQty { get; set; }
        public decimal SaleTotal { get; set; }
        public int NetQty { get; set; }
        public decimal NetTotal { get; set; }
    }
}
